mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/zk/ZookeeperLeaderElector.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2024 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.master.zk;
+
+import static io.mantisrx.shaded.org.apache.zookeeper.KeeperException.Code.OK;
+
+import io.mantisrx.server.core.BaseService;
+import io.mantisrx.server.core.ILeadershipManager;
+import io.mantisrx.server.core.zookeeper.CuratorService;
+import io.mantisrx.server.master.config.ConfigurationProvider;
+import io.mantisrx.shaded.com.fasterxml.jackson.databind.ObjectMapper;
+import io.mantisrx.shaded.org.apache.curator.framework.CuratorFramework;
+import io.mantisrx.shaded.org.apache.curator.framework.recipes.leader.LeaderLatch;
+import io.mantisrx.shaded.org.apache.curator.framework.recipes.leader.LeaderLatchListener;
+import io.mantisrx.shaded.org.apache.zookeeper.CreateMode;
+import io.mantisrx.shaded.org.apache.zookeeper.data.Stat;
+import io.netty.util.concurrent.DefaultThreadFactory;
+import java.io.IOException;
+import java.util.concurrent.Executors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ZookeeperLeaderElector extends BaseService {
```

Comment:
The original purpose was name change only, but that did grow as the `CuratorService` point of view required modifications to the constructor. I removed the builder model as I felt the factory eliminated the need for it. I did not see wide usage of that pattern and I left `LeaderElector` if users still want it.